# Alternate Protocol 1: Predicting ELMs in sequences


TODO: DEDUPLICATE TEXT (COMPARE TO 4-P53_BASIC_PROTOCOL_2.MD)

TODO: DESCRIBE MOST PROBABLE MOTIF INSTANCES (COMPARED TO FILTERED)

We will use protein "CV_0974" (uniprot ID: Q7NZE8) as an example, a "probable
tyrosine phosphatase" from _Chromobacterium violaceum_. This protein is predicted
to be a tyrosine phosphates because it has a "tyrosine phosphatase" (PTPc) domain.

## Necessary Resources

### Software & Hardware

A modern browser such as Firefox, Chrome, Safari.
ELM is best viewed on a laptop or desktop computer, although tablets and
smartphones will also work.

## Submitting a query to ELM

![](Figures/BACT_1/elm_search.png)
**Figure BACT-BP-1:** The input query page for finding motifs in ELM. The
sequence for _C. vilaceum protein_ CV_0974 was used as an example for this
protocol.

step 1. Click on the "ELM Predictions" button in the menu to access the search query
   page (Fig. BACT-BP-1). Here you can provide either a protein accession (from
   uniprot) or an amino acid sequence (simply the
   sequence, or a FASTA formatted entry) in which you want to detect SLiMs.
   Retrieve the FASTA formatted sequence from Uniprot
   (http://www.uniprot.org/uniprot/Q7NZE8.fasta), and enter it into the "sequence
   input text box".

TODO: MENTION NOT TO USE "CHROMOBACTERIUM VIOLACEUM" IN THE ORGANISM BOX AND WHY

![](Figures/BACT_1/elm_results_summary.png)
**Figure BACT-BP-2:** The graphical results summary of the ELM Prediction pipeline
for Probable Tyrosine phosphate (CV_0974). Note that not all motif
detections are shown (the image is truncated at the bottom). The top five rows show a handfull of
structural features. The motif occurence are shown as blue boxes, the intensity of which
indicates the conservation score. See steps XXX to YYY for more information.

step 2. The Results are summarized in the first figure on the results page (see
   figure BACT-BP-2). The Graphical summary shows all of the final and
   intermediate results generated by the ELM Prediction pipeline, and can be
   used infer whether or not a motif is present in a sequence, as well as now
   likely it is to be functional based on its structural context and
   evolutionary conservation.

step 3. Check the first row to see whether there are for phosphorylation sites acid is a serine, threonine
   or tyrosine. In this case, no phosphorylation data could be found in the Phospho.ELM database (PMID:21062810).

step 4. Check the second row showing SMART and Pfam domains. Hover the mouse over these domains to
   see their names and exact start and end positions.

step 5.  The third row shows globular and disordered regions in the sequence as
    predicted by GlobPlot (PMID:12824398). The 4th & 5th
    rows contain results from IUPred (PMID:15955779), another unstructured region
    prediction tool. Protein segments with an IUpred score above 0.5 are 95% likely
    to be disorered (REF???).

step 6. Place the cursor over the blue box for motif occurence
   "DOC_USP7_MATH_1" at position 129-133. This motif is in a disorered region, and
   has not been filtered out by the structural filter. However, its
   conservation score is extremely low: 0.000, indicating it is not conserved
   in homologous proteins. Place the cursor over motif "DOC_MAPK_DCC_7" at
   positions "334-343". Despite the high conservation score (1.000), this motif
   is inside the PTPc domain (and a Globular regions), and therefore has been
   filtered out.

TODO: CHECK CONSERVATION FILTER

## Interpreting the prediction results: Additional Information

TODO: DESCRIBE HOW TO INTERPRETE THE PREDICTIONS USING THIS BACTERIAL EXAMPLE
(OF WHICH NOT MUCH IS KNOWN). FOCUS ON HOW ONE SHOULD INTERPRETE THESE
PREDICTIONS (LOOK AT DISORDER/GLOBULARITY, CONSERVATION)
